File: javalin/src/test/java/io/javalin/examples/FileUploadExample.java
/*
 * Javalin - https://javalin.io
 * Copyright 2017 David Åse
 * Licensed under Apache 2.0: https://github.com/tipsy/javalin/blob/master/LICENSE
 */

package io.javalin.examples;

import io.javalin.Javalin;
import io.javalin.util.FileUtil;
import static io.javalin.testing.JavalinTestUtil.*;

public class FileUploadExample {

    public static void main(String[] args) {

        Javalin app = Javalin.create().start(7000);

        get(app, "/", ctx ->
            ctx.html(
                ""
                    + "<form method='post' enctype='multipart/form-data'>"
                    + "    <input type='file' name='files' multiple>"
                    + "    <button>Upload</button>"
                    + "</form>"
            )
        );

        post(app, "/", ctx -> {
            ctx.uploadedFiles("files").forEach(file -> {
                FileUtil.streamToFile(file.content(), "upload/" + file.filename());
            });
        });

    }

}

File: javalin/src/test/kotlin/io/javalin/examples/FileUploadExample.kt
/*
 * Javalin - https://javalin.io
 * Copyright 2017 David Åse
 * Licensed under Apache 2.0: https://github.com/tipsy/javalin/blob/master/LICENSE
 */

package io.javalin.examples

import io.javalin.Javalin
import io.javalin.util.FileUtil

fun main() {

    Javalin.create {
        it.routes.get("/") { ctx ->
            ctx.html(
                """
                    <form method='post' enctype='multipart/form-data'>
                        <input type='file' name='files' multiple>
                        <button>Upload</button>
                    </form>
                """
            )
        }
        it.routes.post("/") { ctx ->
            ctx.uploadedFiles("files").forEach {
                FileUtil.streamToFile(it.content(), "upload/${it.filename()}")
            }
        }
    }.start(7070)

}
